Apps/US/HybridGP_US/app/src/Codeunits/GPCloudMigrationUS.Codeunit.al

```diff
@@ -30,6 +45,22 @@ codeunit 42004 "GP Cloud Migration US"
         SetPreferredVendorBankAccountsUseForElectronicPayments();
     end;
 
+    local procedure EnsureSupportedReportingYear()
+    var
+        GPCompanyAdditionalSettings: Record "GP Company Additional Settings";
+        GPVendor1099MappingHelpers: Codeunit "GP Vendor 1099 Mapping Helpers";
+        CurrentYear: Integer;
+    begin
+        GPCompanyAdditionalSettings.GetSingleInstance();
+        CurrentYear := System.Date2DMY(Today(), 3);
+
+        // If the configured tax year is less than the minimum supported year (example: 0), default it to the current year
```

Comment:
Should we throw an error instead? Ideally the UI should throw an error and roll back to the current year.

Reply:
I have been working on this feature for some time, and I don't recall why I removed this code... I added it back with this commit. This condition should never happen, the UI does throw an error in this case. There was an early bug where the error wasn't thrown, which is why I think I added this 'fall back' option in the first place. The migration has succeeded by this point, and setting the 1099 year to the current year is a good option as a fall back.